File: level-4/escape-pods/Solution.java
public class Solution {
    
    public static int solution(int[] entrances, int[] exits, int[][] path) {
        int count = 0;
        for (int i = entrances.length; i < path.length - exits.length; i++) {
            int maxPassengers = 0;
            for (int bunnies : path[i])
                maxPassengers += bunnies;

            int availablePassengers = 0;
            for (int j = 0; j < entrances.length; j++)
                availablePassengers += path[j][i];

            count += Math.min(maxPassengers, availablePassengers);
        }
        return count;
    }
}

File: level-4/escape-pods/solution.py
def solution(entrances, exits, path):
    count = 0

    for i in range(len(path) - len(exits)):
        max_passengers = sum(path[i + len(entrances)])
        available_passengers = 0
        
        for j in entrances:
            available_passengers += path[j][i + len(entrances)]

        count += min(available_passengers, max_passengers)
        
    return count
